src/utils/actions.js

```diff
@@ -50,20 +50,23 @@ const _paginatedNetworkFetch = async (url, baseUrl, req, parse) => {
 
 
 const _networkAction = (fn, ...args) => async (dispatch, getState) => {
-    let fnResult = fn(...args);
+    let fnResult = await fn(...args);
     if (typeof fnResult === "function") {
         // Needs dispatch / getState, resolve those.
-        fnResult = fnResult(dispatch, getState);
+        fnResult = await fnResult(dispatch, getState);
     }
 
     const {types, params, url, baseUrl, req, err, onSuccess, onError, paginated} = fnResult;
 
     // Only include access token when we are making a request to this Bento node or the IdP!
     // Otherwise, we could leak it to external sites.
 
-    const token = (url.startsWith("/") || url.startsWith(BENTO_URL) || url.startsWith(IDP_BASE_URL))
-        ? getState().auth.accessToken
-        : null;
+    const token = (
+        url.startsWith("/") ||
+        (BENTO_URL !== null && url.startsWith(BENTO_URL)) ||
+        (BENTO_PUBLIC_URL !== null && url.startsWith(BENTO_PUBLIC_URL)) ||
+        (IDP_BASE_URL !== null && url.startsWith(IDP_BASE_URL))
```

Comment:
I think falsy checks would be better here. `undefined !== null` returns `true`, and according to the [startsWith](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith#parameters) doc:

> All values that are not regexes are [coerced to strings](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String#string_coercion), so omitting it or passing undefined causes startsWith() to search for the string "undefined", which is rarely what you want.